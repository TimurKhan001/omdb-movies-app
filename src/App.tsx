import { BrowserRouter, Route, Routes } from 'react-router-dom'

import FavoritesDisplay from './compoments/Favorites/FavoritesDisplay'
import MovieDetail from './compoments/MovieDetail/MovieDetailContainer'
import MovieSearchContainer from './compoments/MovieSearch/MovieSearchContainer'
import Navigation from './compoments/Navigation'

const App = () => {
  return (
    <BrowserRouter>
      <Navigation />
      <Routes>
        <Route path="/" element={<MovieSearchContainer />} />
        <Route path="/detail/:movieId" element={<MovieDetail />} />
        <Route path="/favorites" element={<FavoritesDisplay />} />
      </Routes>
    </BrowserRouter>
  )
}

export default App
