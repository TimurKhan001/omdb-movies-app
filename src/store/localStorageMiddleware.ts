import type { Middleware } from '@reduxjs/toolkit'

import { saveState } from '../utils/localStorageUtils'

export const localStorageMiddleware: Middleware =
  (store) => (next) => (action) => {
    const result = next(action)
    const favorites = store.getState().favorites.favorites
    saveState('favoritesMovies', favorites)
    return result
  }
