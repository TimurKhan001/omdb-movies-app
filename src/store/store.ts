import { configureStore } from '@reduxjs/toolkit'
import { setupListeners } from '@reduxjs/toolkit/query'

import { localStorageMiddleware } from './localStorageMiddleware'
import appDataReducer from './slices/appData'
import favoritesReducer from './slices/favoritesSlice'
import { movieApi } from './slices/movieApi'
import movieReducer from './slices/movieSlice'
import searchResultsReducer from './slices/searchResultsSlice'

export type RootState = ReturnType<typeof store.getState>

export const store = configureStore({
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware()
      .concat(movieApi.middleware)
      .concat(localStorageMiddleware),
  reducer: {
    appData: appDataReducer,
    [movieApi.reducerPath]: movieApi.reducer,
    favorites: favoritesReducer,
    movie: movieReducer,
    searchResults: searchResultsReducer,
  },
})

export type AppDispatch = typeof store.dispatch

setupListeners(store.dispatch)
