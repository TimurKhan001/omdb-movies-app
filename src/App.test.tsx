import { render, fireEvent } from '@testing-library/react'
import { Provider } from 'react-redux'
import { expect, describe, it } from 'vitest'

import App from './App'
import { store } from './store/store'

describe('App Component', () => {
  // it('renders correctly', () => {
  //   const result = render(
  //     <Provider store={store}>
  //       <App />
  //     </Provider>,
  //   )

  //   expect(result).toMatchSnapshot()
  // })

  it('toggles dark mode correctly', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <App />
      </Provider>,
    )

    const darkModeToggle = getByTestId('dark-mode-toggle')

    expect(darkModeToggle).toBeVisible()

    expect(document.documentElement).not.toHaveClass('dark')

    fireEvent.click(darkModeToggle)
    expect(document.documentElement).toHaveClass('dark')

    fireEvent.click(darkModeToggle)
    expect(document.documentElement).not.toHaveClass('dark')
  })
})
