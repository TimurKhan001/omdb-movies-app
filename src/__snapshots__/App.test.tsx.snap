// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="bg-custom-light-background dark:bg-netflix-dark-background p-4"
      >
        <ul
          class="flex justify-between items-center text-custom-light-text dark:text-netflix-dark-text"
        >
          <div
            class="flex space-x-6"
          >
            <li>
              <a
                class="hover:underline"
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                class="hover:underline"
                href="/favorites"
              >
                Favorites
              </a>
            </li>
          </div>
          <div
            class="flex row-auto justify-start align-baseline"
          >
            <span
              class="mr-4"
            >
              Light
            </span>
            <div
              aria-pressed="false"
              class="cursor-pointer w-12 h-6 flex items-center rounded-full p-1 bg-gray-300"
              data-testid="dark-mode-toggle"
              role="button"
              tabindex="0"
            >
              <div
                class="transition-all duration-300 ease-in-out transform scale-100 bg-white w-5 h-5 rounded-full translate-x-0"
              />
            </div>
            <span
              class="ml-4"
            >
              Dark
            </span>
          </div>
        </ul>
      </nav>
      <div
        class="space-y-2 p-6 dark:text-gray-300"
      >
        <label
          class="block text-gray-500 dark:text-gray-400 text-sm"
          for="searchTerm"
        >
          Search for a movie:
        </label>
        <form
          class="flex flex-col md:flex-row space-y-2 md:space-y-0 md:space-x-4"
        >
          <input
            class="px-2 py-1 border rounded dark:bg-gray-700 dark:border-gray-600 dark:text-white"
            id="searchTerm"
            pattern=".{3,}"
            placeholder="Enter movie title"
            title="Please enter at least 3 characters."
            type="text"
            value=""
          />
          <button
            class="py-1 px-2 bg-blue-500 text-white rounded hover:bg-blue-600 dark:bg-red-500 dark:hover:bg-red-600"
            type="submit"
          >
            Search
          </button>
        </form>
        <div
          class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-5 gap-4 py-12"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <nav
      class="bg-custom-light-background dark:bg-netflix-dark-background p-4"
    >
      <ul
        class="flex justify-between items-center text-custom-light-text dark:text-netflix-dark-text"
      >
        <div
          class="flex space-x-6"
        >
          <li>
            <a
              class="hover:underline"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              class="hover:underline"
              href="/favorites"
            >
              Favorites
            </a>
          </li>
        </div>
        <div
          class="flex row-auto justify-start align-baseline"
        >
          <span
            class="mr-4"
          >
            Light
          </span>
          <div
            aria-pressed="false"
            class="cursor-pointer w-12 h-6 flex items-center rounded-full p-1 bg-gray-300"
            data-testid="dark-mode-toggle"
            role="button"
            tabindex="0"
          >
            <div
              class="transition-all duration-300 ease-in-out transform scale-100 bg-white w-5 h-5 rounded-full translate-x-0"
            />
          </div>
          <span
            class="ml-4"
          >
            Dark
          </span>
        </div>
      </ul>
    </nav>
    <div
      class="space-y-2 p-6 dark:text-gray-300"
    >
      <label
        class="block text-gray-500 dark:text-gray-400 text-sm"
        for="searchTerm"
      >
        Search for a movie:
      </label>
      <form
        class="flex flex-col md:flex-row space-y-2 md:space-y-0 md:space-x-4"
      >
        <input
          class="px-2 py-1 border rounded dark:bg-gray-700 dark:border-gray-600 dark:text-white"
          id="searchTerm"
          pattern=".{3,}"
          placeholder="Enter movie title"
          title="Please enter at least 3 characters."
          type="text"
          value=""
        />
        <button
          class="py-1 px-2 bg-blue-500 text-white rounded hover:bg-blue-600 dark:bg-red-500 dark:hover:bg-red-600"
          type="submit"
        >
          Search
        </button>
      </form>
      <div
        class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-5 gap-4 py-12"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
